packages/graphql-transformers-e2e-tests/src/__tests__/SubscriptionsWithAuthTest.e2e.test.ts

```diff
@@ -0,0 +1,919 @@
+    observer.subscribe({
+        error: (err: any) => {
+            console.log(err.graphQLErrors[0])
+            expect(err.graphQLErrors[0].message).toEqual('Not Authorized to access onCreateStudent on type Subscription')
+            expect(err.graphQLErrors[0].errorType).toEqual('Unauthorized')
+            done()
+        }
+    });
+    await new Promise((res) => setTimeout(() => res(), 1000))
+
+    createStudent(GRAPHQL_CLIENT_1,
+        {
+            name: "student2",
+            email: "student2@domain.com",
+            ssn: "BBB-00-SNSN"
+        })
+})
+
+test('Test a subscription on update', async done => {
+    // susbcribe to update students as user 2
+    const observer = onUpdateStudent(GRAPHQL_CLIENT_2)
+    let subscription = observer.subscribe( (event: any) => {
+        const student = event.data.onUpdateStudent;
+        subscription.unsubscribe()
+        expect(student.id).toEqual(student3ID)
+        expect(student.name).toEqual('student3')
+        expect(student.email).toEqual('emailChanged@domain.com')
+        expect(student.ssn).toBeNull()
+        done();
+    });
+
+    const student3 = await createStudent(GRAPHQL_CLIENT_1,
+        {
+            name: "student3",
+            email: "changeThisEmail@domain.com",
+            ssn: "CCC-01-SNSN"
+        })
+    expect(student3.data.createStudent).toBeDefined()
+    const student3ID = student3.data.createStudent.id
+    expect(student3.data.createStudent.name).toEqual('student3')
+    expect(student3.data.createStudent.email).toEqual('changeThisEmail@domain.com')
+    expect(student3.data.createStudent.ssn).toBeNull()
+
+    updateStudent(GRAPHQL_CLIENT_1,
+        {
+            id: student3ID,
+            email: 'emailChanged@domain.com'
+        })
+})
+
+
+test('Test a subscription on delete', async done => {
+    // subscribe to onDelete as user 2
+    const observer = onDeleteStudent(GRAPHQL_CLIENT_2)
+    let subscription = observer.subscribe( (event: any) => {
+        const student = event.data.onDeleteStudent;
+        subscription.unsubscribe()
+        expect(student.id).toEqual(student4ID)
+        expect(student.name).toEqual('student4')
+        expect(student.email).toEqual('plsDelete@domain.com')
+        expect(student.ssn).toBeNull()
+        done();
+    });
+
+    const student4 = await createStudent(GRAPHQL_CLIENT_1,
+        {
+            name: "student4",
+            email: "plsDelete@domain.com",
+            ssn: "DDD-02-SNSN"
+        })
+    expect(student4).toBeDefined()
+    const student4ID = student4.data.createStudent.id
+    expect(student4.data.createStudent.email).toEqual('plsDelete@domain.com')
+    expect(student4.data.createStudent.ssn).toBeNull()
+
+    await deleteStudent(GRAPHQL_CLIENT_1, { id: student4ID })
+})
+
+// ownerField Tests
+test('Test subscription onCreatePost with ownerField', async done => {
+    const observer = GRAPHQL_CLIENT_1.subscribe({ query: gql`
```

Comment:
Why different pattern compared to student test. I see a helper called `onCreateStudent` but no helper here. I like the code without helper as that gives more idea whats happening in the test

Reply:
Moved Subscription queries within the tests for clarity.